lib/portage/gpkg.py

```diff
@@ -1081,35 +1097,38 @@ def update_metadata(self, metadata, newcpv=None):
             name=tmp_gpkg_file_name, mode="w", format=container_tar_format
         ) as container:
             # gpkg version
-            gpkg_version_file = tarfile.TarInfo(self.gpkg_version)
+            gpkg_version_file = tarfile.TarInfo(
+                os.path.join(new_basename, self.gpkg_version)
+            )
             gpkg_version_file.mtime = datetime.utcnow().timestamp()
             container.addfile(gpkg_version_file)
+            checksum_info = checksum_helper(self.settings)
+            checksum_info.finish()
+            self._record_checksum(checksum_info, gpkg_version_file)
 
             compression_cmd = self._get_compression_cmd()
 
             # metadata
-            if newcpv:
-                self.base_name = newcpv
-                self._add_metadata(container, metadata, compression_cmd)
-                self.base_name = oldcpv
-            else:
-                self._add_metadata(container, metadata, compression_cmd)
+            self._add_metadata(container, metadata, compression_cmd)
 
             # reuse other stuffs
             with tarfile.open(self.gpkg_file, "r") as container_old:
                 manifest_old = self.manifest_old.copy()
 
                 for m in manifest_old:
                     file_name_old = m[1]
+                    if os.path.basename(file_name_old) == self.gpkg_version:
+                        continue
                     if os.path.basename(file_name_old).startswith("metadata"):
                         continue
-                    old_data_tarinfo = container_old.getmember(file_name_old)
+                    old_data_tarinfo = container_old.getmember(
+                        os.path.join(self.prefix, file_name_old)
+                    )
                     new_data_tarinfo = copy(old_data_tarinfo)
-                    if newcpv:
-                        m[1] = m[1].replace(oldcpv, newcpv, 1)
-                        new_data_tarinfo.name = new_data_tarinfo.name.replace(
-                            oldcpv, newcpv, 1
-                        )
+                    m[1] = m[1].replace(old_basename, new_basename, 1)
```

Comment:
Is this still relevant now that Manifest doesn't contain full paths?

Reply:
No, it is no longer needed.